Update upgrade guide (#5824)
cupy/cupy: docs/source/upgrade.rst

CuPy v10
========

Dropping CUDA 9.2 / 10.0 / 10.1 Support
---------------------------------------

CUDA 10.1 or earlier is no longer supported.

> Use CUDA 9.2 or later.

Reviewer: ```suggestion
Use CUDA 10.2 or later.
```

NCCL and cuDNN shared libraires are no longer included in wheels (see `#4850 <https://github.com/cupy/cupy/issues/4850>`_ for discussions). 
You can manually install them after installing wheel if you don't have a previous installation; see :doc:`install` for details.

cuTENSOR Enabled in Wheels
--------------------------

cuTENSOR can now be used when installing CuPy via wheels.

``cupy.cuda.{nccl,cudnn}`` Modules Needs Explicit Import
--------------------------------------------------------

> Previously ``import cupy.cuda`` automatically made ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules available.

Reviewer: ```suggestion
Previously ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules were available when directly importing CuPy with ``import cupy.cuda``
```

Author: To be more precise, just doing `import cupy` made these modules available.
```suggestion
Previously ``cupy.cuda.nccl`` and ``cupy.cuda.cudnn`` modules were automatically imported.
```

Reviewer: yeah, this sounds much better :)

> In CuPy v9, you need to import them explicitly to use these modules (i.e., ``import cupy.cuda.nccl`` / ``import cupy.cuda.cudnn``.)

Reviewer: ```suggestion
In CuPy v9, these modules need to be explicitly imported (i.e., ``import cupy.cuda.nccl`` / ``import cupy.cuda.cudnn``.)
```
